capa/main.py

```diff
@@ -597,20 +627,24 @@ def get_rules(rule_paths: List[str], disable_progress=False) -> List[Rule]:
         # to disable progress completely
         pbar = lambda s, *args, **kwargs: s
 
-    for rule_file_path in pbar(list(rule_file_paths), desc="loading ", unit=" rules"):
+    for path, content in pbar(zip(rule_file_paths, rule_contents), desc="loading ", unit=" rules"):
         try:
-            rule = capa.rules.Rule.from_yaml_file(rule_file_path)
+            rule = capa.rules.Rule.from_yaml(content.decode("utf-8"))
         except capa.rules.InvalidRule:
             raise
         else:
-            rule.meta["capa/path"] = rule_file_path
-            if is_nursery_rule_path(rule_file_path):
+            rule.meta["capa/path"] = path
+            if is_nursery_rule_path(path):
                 rule.meta["capa/nursery"] = True
 
             rules.append(rule)
             logger.debug("loaded rule: '%s' with scope: %s", rule.name, rule.scope)
 
-    return rules
+    ruleset = capa.rules.RuleSet(rules)
+
+    capa.rules.cache.cache_ruleset(cache_dir, ruleset)
```

Comment:
should this happen in here? I could confused in `cache-ruleset` where it writes the cache
maybe we can at least document it in the function

Reply:
good point, this function definitely needs a docstring.

the function `cache_ruleset` writes the cache to disk, using the best determined cache directory. this code is reached in source distributions of capa or when the rules directory is provided via the cli flag.

the script `cache-ruleset.py` is only used in CI to pre-build a cache to be embedded within the PyInstaller bundle.

Reply:
[e30dd08](https://github.com/mandiant/capa/pull/1291/commits/e30dd08dec2f3099ea66bf189a1a77651d4e7fa3)

please take another look and let me know what is not very clear.

Reply:
Thanks, this is good. I wasn't fully clear as I meant a doc on `get_rules` that it saves the rule cache.

Reply:
related: what if callers don't want to save the rule cache?

Reply:
i think its feasible to support an environment variable that might disable this caching behavior; however, it would introduce a bit more code. i figured unless we hear that users need this, we should wait to add the additional complexity.

Reply:
for example, i can imagine a user that runs capa within a docker container and/or restricted host that doesn't want a lot of file system activity. they might want to turn off the cache, especially if they hit permission errors.

Reply:
makes sense, thanks!